relm4-macros/src/attrs.rs

```diff
@@ -1,84 +1,111 @@
-use proc_macro2::TokenStream as TokenStream2;
-use quote::ToTokens;
+use proc_macro2::Span;
 use syn::Ident;
+use syn::Path;
+use syn::PathArguments;
+use syn::PathSegment;
 use syn::Result;
 use syn::Token;
 use syn::Visibility;
+// use syn::buffer::Cursor;
 use syn::parse::Parse;
 use syn::parse::ParseStream;
+use syn::punctuated::Punctuated;
+use syn::token::Colon2;
 
 #[derive(Debug)]
-pub struct WidgetVisibility {
-    pub_vis: Option<Visibility>,
-}
-
-impl WidgetVisibility {
-    fn new() -> Self {
-        WidgetVisibility{
-            pub_vis: None,
-        }
-    }
-}
-
-impl ToTokens for WidgetVisibility {
-    fn to_tokens(&self, out: &mut TokenStream2) {
-        if let Some(vis_pub) = &self.pub_vis {
-            vis_pub.to_tokens(out);
-        }
-    }
-}
-
 pub struct Attrs {
     /// Keeps information about visibility of the widget
-    pub visibility: WidgetVisibility,
+    pub visibility: Option<Visibility>,
```

Comment:
I think `visibility_set` can be removed and `visibility.is_some()` can be used instead.